saw-core/src/SAWCore/Typechecker.hs

```diff
@@ -421,10 +421,10 @@ processDecls (Un.DataDecl (PosPair p nm) param_ctx dt_tp c_decls : rest) =
     mapM (\(Un.Ctor (PosPair p' c) ctx body) ->
            (c,) <$> typeInferComplete (Un.Pi p' ctx body)) c_decls
   ctors <-
-    case ctxBindingsOfTerms dtParams of
-      ExistsTp p_ctx ->
-        case ctxBindingsOfTerms dtIndices of
-          ExistsTp ix_ctx ->
+    case dtParams of
+      p_ctx ->
+        case dtIndices of
+          ix_ctx ->
```

Comment:
These `case` expressions don't anything except bind the names `dtParams`/`dtIndices` to `p_ctx`/`ix_ctx`. I'd suggest either converting these to `let`-bindings (e.g., `let p_ctx = dtParams in ...`) or just removing them entirely to simplify the code.

Reply:
Fixed in 524244ed623ffef8f2a43c5c22499d2e552ef68d.